packages/flutter/lib/src/widgets/widget_span.dart

```diff
@@ -90,17 +90,26 @@ class WidgetSpan extends PlaceholderSpan {
   /// Helper function for extracting [WidgetSpan]s in preorder, from the given
   /// [InlineSpan] as a list of widgets.
   ///
-  /// The `textScaleFactor` is the the number of font pixels for each logical
-  /// pixel.
+  /// The `textScaler` is the scaling strategy for scaling the content.
   ///
   /// This function is used by [EditableText] and [RichText] so calling it
   /// directly is rarely necessary.
-  static List<Widget> extractFromInlineSpan(InlineSpan span, double textScaleFactor) {
+  static List<Widget> extractFromInlineSpan(InlineSpan span, TextScaler textScaler) {
     final List<Widget> widgets = <Widget>[];
+    final List<double> fontSizeStack = <double>[14.0];
     int index = 0;
     // This assumes an InlineSpan tree's logical order is equivalent to preorder.
-    span.visitChildren((InlineSpan span) {
+    bool visitSubtree(InlineSpan span) {
+      final double? fontSize = switch (span.style?.fontSize) {
+        final double size when size != fontSizeStack.last => size,
+        _ => null,
+      };
+      if (fontSize != null) {
+        fontSizeStack.add(fontSize);
+      }
```

Comment:
Isnt' this simply:

```dart
final double? fontSize = span.style?.fontSize;
if (fontSize != null && fontSize != fontSizeStack.last) {
  fontSizeStack.add(fontSize);
}
```

Reply:
This is pushing the fontSize to the stack. Whether `fontSize` is null also indicates if a pop operation is needed at the end. I guess I'll rename the variable to fontSizeToPush